### PR TITLE
For #24495 fix and re-enable settingsPrivacyItemsTest UI test.

### PR DESCRIPTION
**Pull request by AndiAJ**
For #24495 fix and re-enable `settingsPrivacyItemsTest` UI test ✅  successfully passed 40x on Firebase

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsRobot.kt [hunk 1]
```diff
@@ -73,6 +75,8 @@ class SettingsRobot {
     // PRIVACY SECTION
     fun verifyPrivacyHeading() = assertPrivacyHeading()
 
+    fun verifyHTTPSOnlyModeButton() = assertHTTPSOnlyModeButton()
+    fun verifyHTTPSOnlyModeValue(state: String) = assertHTTPSOnlyModeValue(state)
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsRobot.kt:79]: I think it's clearer if we rename this function to `verifyHTTPSOnlyModeState`

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsRobot.kt [hunk 2]
```diff
@@ -73,6 +75,8 @@ class SettingsRobot {
     // PRIVACY SECTION
     fun verifyPrivacyHeading() = assertPrivacyHeading()
 
+    fun verifyHTTPSOnlyModeButton() = assertHTTPSOnlyModeButton()
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsRobot.kt:78]: Probably clearer if we rename this function to `verifyHTTPSOnlyModeToggle` - it's a toggle right?

**Review by AaronMT — CHANGES_REQUESTED** [review]
Just some minor function rename

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsRobot.kt [hunk 3]
```diff
@@ -429,8 +449,12 @@ private fun assertDeleteBrowsingDataOnQuitButton() {
 }
 
 private fun assertDeleteBrowsingDataValue(state: String) {
-    mDevice.wait(Until.findObject(By.text("Delete browsing data on quit")), waitingTime)
-    onView(withText(state)).check(matches(withEffectiveVisibility(Visibility.VISIBLE)))
+    onView(
```
Comment (sv-ohorvath) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsRobot.kt:452]: same, about renaming the methods above: let's apply it here then too

**Review by sv-ohorvath — APPROVED** [review]

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsRobot.kt (hunk 2 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsRobot.kt:78]: No, it's not a toggle, it's a button in "Settings"
The other items have the same naming.
image

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsRobot.kt (hunk 1 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsRobot.kt]: Done!

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsRobot.kt (hunk 3 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsRobot.kt:452]: Done!

**Review by AaronMT — APPROVED** [review]

**Pull request merged**